Source/ThirdParty/libwebrtc/Source/webrtc/sdk/objc/components/video_codec/nalu_rewriter.cc

```diff
@@ -467,6 +468,236 @@ CMVideoFormatDescriptionRef CreateVideoFormatDescription(
+        // cpb_cnt_minus1
+        auto cpb_cnt_minus1 = reader.ReadExponentialGolomb();
+        // bit_rate_scale and bit_rate_scale
+        reader.ConsumeBits(8);
+        for (size_t cptr = 0; cptr < cpb_cnt_minus1; ++cptr) {
+            // bit_rate_value_minus1
+            reader.ReadExponentialGolomb();
+            // cpb_size_value_minus1
+            reader.ReadExponentialGolomb();
+            // cbr_flag
+            reader.ConsumeBits(1);
+        }
+    }
+};
+
+// Table A-1 of H.264 spec
+static size_t maxDpbMbsFromLevelNumber(uint8_t profile_idc, uint8_t level_idc, bool constraint_set3_flag)
+{
+  if ((profile_idc == 66 || profile_idc == 77) && level_idc == 11 && constraint_set3_flag) {
+    // level1b
+    return 396;
+  }
+  H264Level level_casted = static_cast<H264Level>(level_idc);
+
+  switch (level_casted) {
+  case H264Level::kLevel1:
+    return 396;
+  case H264Level::kLevel1_1:
+    return 900;
+  case H264Level::kLevel1_2:
+  case H264Level::kLevel1_3:
+  case H264Level::kLevel2:
+    return 2376;
+  case H264Level::kLevel2_1:
+    return 4752;
+  case H264Level::kLevel2_2:
+  case H264Level::kLevel3:
+    return 8100;
+  case H264Level::kLevel3_1:
+    return 18000;
+  case H264Level::kLevel3_2:
+    return 20480;
+  case H264Level::kLevel4:
+    return 32768;
+  case H264Level::kLevel4_1:
+    return 32768;
+  case H264Level::kLevel4_2:
+    return 34816;
+  case H264Level::kLevel5:
+    return 110400;
+  case H264Level::kLevel5_1:
+    return 184320;
+  case H264Level::kLevel5_2:
+    return 184320;
+  default:
+    RTC_LOG(LS_ERROR) << "Wrong maxDpbMbsFromLevelNumber";
+    return 0;
+  }
+}
+
+uint64_t ComputeH264ReorderSize(const uint8_t* annexb_buffer,
+                                size_t annexb_buffer_size) {
+  AnnexBBufferReader reader(annexb_buffer, annexb_buffer_size);
+  if (!reader.SeekToNextNaluOfType(kSps)) {
+    return 0;
+  }
+  const uint8_t* spsData;
+  size_t spsDataSize;
+
+  if (!reader.ReadNalu(&spsData, &spsDataSize)) {
+    return 0;
+  }
+
+  std::vector<uint8_t> unpacked_buffer = H264::ParseRbsp(spsData, spsDataSize);
+  auto spsAndVui = SpsAndVuiParser::Parse(unpacked_buffer);
+  if (!spsAndVui) {
+    RTC_LOG(LS_ERROR) << "Failed to parse sps.";
+    return 0;
+  }
+
+  if (spsAndVui->pic_order_cnt_type == 2) {
+    return 0;
+  }
+
+  uint64_t max_dpb_mbs = maxDpbMbsFromLevelNumber(spsAndVui->profile_idc, spsAndVui->level_idc, spsAndVui->constraint_set3_flag);
+  uint64_t max_dpb_frames = max_dpb_mbs / ((spsAndVui->width / 16) * (spsAndVui->height / 16));
+  max_dpb_frames = std::min(max_dpb_frames, 16ull);
+
+  if (spsAndVui->bitstream_restriction_flag) {
+    return std::min(spsAndVui->max_num_reorder_frames, max_dpb_frames);
+  }
+  if (spsAndVui->constraint_set3_flag && (spsAndVui->profile_idc == 44 || spsAndVui->profile_idc == 86 || spsAndVui->profile_idc == 100 || spsAndVui->profile_idc == 110 || spsAndVui->profile_idc == 122 || spsAndVui->profile_idc == 244)) {
+    return 0;
```

Comment:
so this is what Chrome does, but I don't see why this would impact the re-ordering queue size:
like profile 44 is 2C which is `C44I` `CAVLC 4:4:4 Intra profile` 

Also, for reading max_num_reorder_frames it is stated:
`When the max_num_reorder_frames syntax element is not present, the value of max_num_reorder_frames value shall be inferred as follows: If profile_idc is equal to 44, 86, 100, 110, 122, or 244 and constraint_set3_flag is equal to 1, the value of max_num_reorder_frames shall be inferred to be equal to 0.`

This isn't what is being done here, it uses the profile_idc value and override the value even when max_num_reorder_frames is defined. 

In any case, `max_num_ref_frames` is the value that should be used, this is what you configure with the encoder . And the only party I see using max_num_reorder_frames is google weirdly enough.

Reply:
The spec defines this rule in different places. Each intra profile is clear about this though so this seems ok, see appendix A.2.9 for instance.